### PR TITLE
SAM-175 Generate set from Data Lined to a Sample Set

### PR DESCRIPTION
**Pull request by charleshtrenholm**

### CODE REVIEW

File: lib/sample_uploader/sample_uploaderImpl.py [hunk 1]
```diff
@@ -813,6 +818,146 @@ def get_sampleset_meta(self, ctx, params):
...
+           "save_date" of string, parameter "version" of int, parameter "saved_by"
+           of string, parameter "ws_id" of int, parameter "workspace" of string,
+           parameter "chsum" of string, parameter "size" of int, parameter
+           "meta" of Mapping of string, string.
+        """
+        # ctx is the context object
+        # return variables are: results
+        #BEGIN create_data_set_from_links
+        sample_service = SampleService(self.sample_url)
+        set_api = SetAPI(self.callback_url)
```
Comment (Tianhao-Gu) [inline lib/sample_uploader/sample_uploaderImpl.py:840]: set_api is never used. we could remove it.

File: lib/sample_uploader/sample_uploaderImpl.py [hunk 2]
```diff
@@ -813,6 +818,146 @@ def get_sampleset_meta(self, ctx, params):
...
+
+            upas = [f"{i[6]}/{i[0]}/{i[4]}" for i in data_objs]
+
+            try:
+                set_obj = {
+                    'description': set_item['description'],
+                    'items': [{'ref': u} for u in upas],
+                }
+
+                save_data = {
```
Comment (Tianhao-Gu) [inline lib/sample_uploader/sample_uploaderImpl.py:921]: `save_data` is also never used.

**Review by Tianhao-Gu — APPROVED** [review]
Overall LGTM 👍 . Just a few unused variables that could be removed.

File: lib/sample_uploader/sample_uploaderImpl.py [hunk 3]
```diff
@@ -813,6 +818,146 @@ def get_sampleset_meta(self, ctx, params):
...
+                    if s['version'] > cache[s['id']]:
+                        cache[s['id']] = s['version']
+
+            sample_ids = [{'id': key, 'version': val} for key, val in cache.items()]
+        except KeyError as e:
+            raise ValueError(
+                f'Invalid sample set ref - sample in dataset missing the "{str(e)}" field'
+            )
+        data_links = sample_service.get_data_links_from_sample_set({
+            'effective_time': now,
```
[the remark this answers is not in the document]
Comment (charleshtrenholm) [reply lib/sample_uploader/sample_uploaderImpl.py:875]: Unfortunately I can't remove the `effective_time` variable without having to change the sample service first. It was on my to do list to fix that but currently it needs an effective time param

[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply lib/sample_uploader/sample_uploaderImpl.py:873]: 👍

**Pull request merged**